pkg/portfwdserver/server.go

```diff
@@ -35,7 +40,23 @@ func (s *TunnelServer) Start(stream api.GuestService_TunnelServer) error {
 		return err
 	}
 	rw := &GRPCServerRW{stream: stream, id: in.Id}
-	bicopy.Bicopy(rw, conn, nil)
+
+	// FIXME: consolidate bicopy and tcpproxy into one
+	//
+	// The bicopy package does not seem to work with `w3m -dump`:
+	// https://github.com/lima-vm/lima/issues/3685
+	//
+	// However, the tcpproxy package can't pass the CI for WSL2 (experimental):
+	// https://github.com/lima-vm/lima/pull/3686#issuecomment-3034842616
+	if wsl2, _ := seemsWSL2(); wsl2 {
+		bicopy.Bicopy(rw, conn, nil)
+	} else {
+		proxy := tcpproxy.DialProxy{DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+			return conn, nil
+		}}
+		proxy.HandleConn(rw)
+	}
```

Comment:
This workaround is quite dirty, but let's merge this and call it a day to unblock the v1.2 release

Reply:
Will open an issue after merging this PR